src/dataregistry/registrar_util.py

```diff
@@ -80,3 +94,58 @@ def get_directory_info(path):
                 num_files += subdir_num_files
                 total_size += subdir_total_size
     return num_files, total_size
+
+def calculate_special(name, v_string, v_suffix, dataset_table, engine):
+    '''
+    Utility to figure out what new version fields should be if caller
+    to register supplies a special version string
+    '''
+    stmt = select(dataset_table.c["version_major","version_minor",
+                                  "version_patch"])\
+                                  .where(dataset_table.c.name == name)
+    if v_suffix:
+        stmt = stmt.where(dataset_table.c.version_suffix == v_suffix)
+        stmt = stmt.order_by(dataset_table.c.version_major.desc())\
```

Comment:
Should these be in this `if`, want to order without suffix too?

Reply:
I'm not sure I understand what you're asking.
`version_suffix` is already constrained to be a single value so it doesn't need an `order_by`

Reply:
I think this file has changed from the comment so we can ignore this